src/project-css/app/Http/Controllers/CollectionController.php

```diff
@@ -51,36 +54,94 @@ public function index() {
     return view('admin/collection')->with('collcntNms', $collcntNms);
   }
 
+  /**
+  * Returns view for collection
+  *
+  * @return view
+  */    
+  public function show($colID) {
+    // find the collection
+    $thisClctn = Collection::findorFail($colID);  
+
+    // Get all the tables of this collection
+    $tbls = $thisClctn->tables()->get();
+
+    // redirect to show page
+    return view('collection/show')->with('tbls', $tbls)
+                                  ->with('colID', $colID)
+                                  ->with('clctnName', $thisClctn->clctnName);
+  }   
+  
+  /**
+  * Returns view for collection
+  *
+  * @return view
+  */    
+  public function upload($colID) {
+    // find the collection
+    $thisClctn = Collection::findorFail($colID);  
+
+    // redirect to show page
+    return view('collection/upload')->with('colID', $colID)
+                                    ->with('clctnName', $thisClctn->clctnName);
+  }   
+
   /**
   * Takes request validates the collection name and saves new collection to database
   */
   public function create(Request $request) {
     // Validate the request before storing the data
     $this->validate($request, $this->rules, $this->messages);
 
-    // Create the collection name
-    $thisClctn = new Collection;
-    $thisClctn->clctnName = $request->clctnName;
-    $thisClctn->save();
+    // Get required fields for collection
```

Comment:
What is the goal with this data, i've seen it define in many duplicate spots like this.  Maybe you need a helper method that takes a parameter and sets the data based on that and returns this information as an array.

Reply:
Added additional description when creating the $data array. It is used to gather sometimes large amount of values so we can easily pass them to helpers.

Reply:
Just to note the use of $data in this fashion is typically of laravel. In some of the auto generated methods for the user this style of passing values to functions is used.